include/albatross/src/models/ransac.hpp

```diff
@@ -100,21 +153,26 @@ ransac(const RansacFunctions<FitType> &ransac_functions,
     if (candidate_consensus.size() >= min_consensus_size) {
       double consensus_metric_value =
           ransac_functions.consensus_metric(candidate_consensus);
-      if (consensus_metric_value < best_consensus_metric) {
-        best_consensus = candidate_consensus;
-        best_consensus_metric = consensus_metric_value;
+      if (consensus_metric_value < output.consensus_metric) {
+        output.inliers = candidate_consensus;
+        output.consensus_metric = consensus_metric_value;
+        output.outliers = outliers;
       }
     }
+
+    ++i;
   }
-  return best_consensus;
+
+  output.return_code = RANSAC_RETURN_CODE_SUCCESS;
```

Comment:
Logically, it looks like if no valid candidate set forms a valid consensus set, then we'll result in setting the return code to `rc_SUCCESS` despite having no `output.inliers`, which downstream will result in an empty fit/crash.

Reply:
yep good catch.